application/src/main/java/uk/nhs/hee/tis/revalidation/repository/RecommendationElasticSearchRepository.java

```diff
@@ -45,6 +45,9 @@ Page<RecommendationView> findByUnderNotice(final String searchQuery,
   Page<RecommendationView> findAll(final String searchQuery,
       final String dbcs, List<String> hiddenGmcIds, final Pageable pageable);
 
+  @Query("{\"bool\":{\"filter\":[{\"match\":{\"underNotice\":\"YES\"}},{\"match\":{\"designatedBody\":\"?2\"}},{\"match_phrase_prefix\":{\"?0\":{\"query\":\"?1\"}}}]}}")
+  List<RecommendationView> autocomplete(final String fieldName, final String input, final String dbcs);
```

Comment:
method name should reflect what it does,
 e.g. `findByFieldNameParameter(String, String, String)`?